src/main/java/teammates/client/scripts/AdminEmailListGenerator.java

```diff
@@ -0,0 +1,484 @@
+            if (account != null){
+                return account.getCreatedAt();
+            }
+        }
+        
+        if(CourseIdToCreatedDateMap.get(student.getCourseId()) != null){
+            return CourseIdToCreatedDateMap.get(student.getCourseId());
+        }
+        
+        Course course = getCourseEntity(student.getCourseId());
+        
+        if(course != null){
+            CourseIdToCreatedDateMap.put(student.getCourseId(), course.getCreatedAt());
+            return course.getCreatedAt();
+        }
+        
+        return null;
+        
+    }
+    
+    private Course getCourseEntity(String courseId){
+        
+        Query q = pm.newQuery(Course.class);
+        q.declareParameters("String courseIdParam");
+        q.setFilter("ID == courseIdParam");
+        
+        @SuppressWarnings("unchecked")
+        List<Course> courseList = (List<Course>) q.execute(courseId);
+        
+        if (courseList.isEmpty() || JDOHelper.isDeleted(courseList.get(0))) {
+            return null;
+        }
+    
+        return courseList.get(0);
+    }
+    
+    private Account getAccountEntity(String googleId) {
+        
+        try {
+            Key key = KeyFactory.createKey(Account.class.getSimpleName(), googleId);
+            Account account = pm.getObjectById(Account.class, key);
+            
+            if (JDOHelper.isDeleted(account)) {
+                return null;
+            } 
+            
+            return account;
+            
+        } catch (IllegalArgumentException iae){
+            return null;            
+        } catch(JDOObjectNotFoundException je) {
+            return null;
+        }
+    }
+    
+    
+    private String getCurrentDateForDisplay(){
+        Date now = new Date();
+        
+        Calendar cal = Calendar.getInstance();
+        cal.clear();
+        cal.setTime(now);
+        cal = TimeHelper.convertToUserTimeZone(cal, Const.SystemParams.ADMIN_TIMZE_ZONE_DOUBLE);
+        
+        System.out.print(formatTime(cal.getTime()));
+        return formatTime(cal.getTime());
+        
+    }
+    
+    private String formatTime(Date date) {
+        if (date == null)
+            return "";
+        return new SimpleDateFormat("[HH-mm]dd-MMM-yyyy]").format(date);
+        
+    }
+    
+    private boolean isValidDate(int day, int month, int year){
+        
+       boolean isDateValid = false; 
+        
+            
+        if(day <= 0 || month <= 0 || year <= 0){
+            isDateValid = false;
+        } else if(day > getMaxNumOfDayForMonth(month, year)){
+            isDateValid = false;
+        } else {
+          isDateValid = true;
+        }
+   
+       if(!isDateValid){
+           System.out.print("Date is not valid. Please Re-enter date.\n\n");
+       } else {
+           System.out.print("Date Entered is valid.\n\n");
+       }
+       
+       return isDateValid;
+        
+    }
+    
+    private int getMaxNumOfDayForMonth(int month, int year){
```

Comment:
If Java doesn't have built-in libraries for this, may be use an external library such as jodatime or natty?